lib/temporal/SQL/update_stds_spatial_temporal_extent_template.sql

```diff
@@ -14,53 +15,58 @@
+       ) AS new_stats
+ WHERE STDS_absolute_time.id = 'SPACETIME_ID';
+
+UPDATE STDS_relative_time
+   SET
+       STDS_relative_time.start_time = new_stats.start_time_new,
+       STDS_relative_time.end_time = new_stats.end_time_new
+  FROM
+       (SELECT
+           min(start_time) AS start_time_new,
+           max(end_time) AS end_time_new
+       FROM GRASS_MAP_relative_time WHERE GRASS_MAP_relative_time.id IN
     		(SELECT id FROM SPACETIME_REGISTER_TABLE)
-       ) WHERE id = 'SPACETIME_ID';
+       ) AS new_stats
+ WHERE STDS_relative_time.id = 'SPACETIME_ID';
+
 -- Update the spatial extent
-UPDATE STDS_spatial_extent SET north =
-       (SELECT max(north) FROM GRASS_MAP_spatial_extent WHERE GRASS_MAP_spatial_extent.id IN
-    		(SELECT id FROM SPACETIME_REGISTER_TABLE)
-       ) WHERE id = 'SPACETIME_ID';
-UPDATE STDS_spatial_extent SET south =
-       (SELECT min(south) FROM GRASS_MAP_spatial_extent WHERE GRASS_MAP_spatial_extent.id IN
-    		(SELECT id FROM SPACETIME_REGISTER_TABLE)
-       ) WHERE id = 'SPACETIME_ID';
-UPDATE STDS_spatial_extent SET east =
-       (SELECT max(east) FROM GRASS_MAP_spatial_extent WHERE GRASS_MAP_spatial_extent.id IN
-    		(SELECT id FROM SPACETIME_REGISTER_TABLE)
-       ) WHERE id = 'SPACETIME_ID';
-UPDATE STDS_spatial_extent SET west =
-       (SELECT min(west) FROM GRASS_MAP_spatial_extent WHERE GRASS_MAP_spatial_extent.id IN
-    		(SELECT id FROM SPACETIME_REGISTER_TABLE)
-       ) WHERE id = 'SPACETIME_ID';
-UPDATE STDS_spatial_extent SET top =
-       (SELECT max(top) FROM GRASS_MAP_spatial_extent WHERE GRASS_MAP_spatial_extent.id IN
-    		(SELECT id FROM SPACETIME_REGISTER_TABLE)
-       ) WHERE id = 'SPACETIME_ID';
-UPDATE STDS_spatial_extent SET bottom =
-       (SELECT min(bottom) FROM GRASS_MAP_spatial_extent WHERE GRASS_MAP_spatial_extent.id IN
-    		(SELECT id FROM SPACETIME_REGISTER_TABLE)
-       ) WHERE id = 'SPACETIME_ID';
-UPDATE STDS_spatial_extent SET proj =
-       (SELECT min(proj) FROM GRASS_MAP_spatial_extent WHERE GRASS_MAP_spatial_extent.id IN
+UPDATE STDS_spatial_extent
+   SET
+       STDS_spatial_extent.north = new_stats.north_new,
+       STDS_spatial_extent.south = new_stats.south_new,
+       STDS_spatial_extent.east = new_stats.east_new,
+       STDS_spatial_extent.west = new_stats.west_new,
+       STDS_spatial_extent.top = new_stats.top_new,
+       STDS_spatial_extent.bottom = new_stats.bottom_new,
+       STDS_spatial_extent.proj = new_stats.proj_new
+       (SELECT
+           max(north) AS north_new,
+           min(south) AS south_new,
+           max(east) AS east_new,
+           min(west) AS west_new,
+           max(top) AS top_new,
+           min(bottom) bottom = bottom_new,
+           min(proj) AS proj_new
+       FROM GRASS_MAP_spatial_extent WHERE GRASS_MAP_spatial_extent.id IN
     		(SELECT id FROM SPACETIME_REGISTER_TABLE)
-       ) WHERE id = 'SPACETIME_ID';
+       ) AS new_stats
+ WHERE STDS_spatial_extent.id = 'SPACETIME_ID';
```

Comment:
I'd be curious to know the difference of using a `WHERE ... IN (SELECT ... ...)` vs a join and appropriate where conditions. 

I know SQLite's `EXPLAIN`/`EXPLAIN QUERY PLAN` are not perfect to show the impact, ie, not like the visual explain of MySql workbench, but it could maybe point something out.

Why I'm asking? In MySql/MariaDB I've never had a good luck with using `IN` like this, it never performed good. It was often changing into joins that it could have a better query plan and not hog up. But since it's a different DBMS, and very different from MySQL in many ways, maybe it doesn't matter.

Reply:
Yes, I was wondering too. I ggogled it and what I understood from SO was that it depends on the particular query and DBMS.
TGIS supports SQLite and PostgrSQL. I guess it does not make adifference for PG but for SQLite maybe....

I switched to using JOIN, but will test what works better ASAp...

Reply:
Thats why I wanted to know if the results of the EXPLAIN QUERY PLAN or EXPLAIN would help enough to know if it would be optimized down to the same.

I think theres a case where it could be disadvantageous, is when using the join, the optimizer then can't use a good (or any index) to help with the join+where of that joined table, and ends up needing more work than with the subquery. If I'm not wrong, a WHERE ... IN subquery ends up equivalent to a for loop, while a successful join would be able to limit the rows to process by using an index or any other trick.

General info:
https://www.geeksforgeeks.org/sql-correlated-subqueries/
https://www.geeksforgeeks.org/sql-join-vs-subquery/
https://www.geeksforgeeks.org/difference-between-nested-subquery-correlated-subquery-and-join-operation/

Reply:
If your database is available locally, I often like to use the open source DBeaver to write and analyze queries (at one point in time I had access to a Devart dbForge licence and always used that). It can connect and manage any DB that can be connected by a JDBC driver. If there's a way to explain the query, it will show you. But you'll probably need to test out pre-replaced queries though. But you'll get timings from the database only, that will limit the impact of other parts of the code.

*I never wrote SQL specific for spatial datasets though

Reply:
Thanks for the feedback, @echoix 

The query plan did not tell me too much. This is the SQLite query plan with `WHERE-IN`:

```
QUERY PLAN
|--MATERIALIZE new_stats
|  |--USE TEMP B-TREE FOR count(DISTINCT)
|  |--SEARCH raster_metadata USING INDEX sqlite_autoindex_raster_metadata_1 (id=?)
|  `--USING INDEX sqlite_autoindex_raster_map_register_0451659289d64936b320841a3c074dda_1 FOR IN-OPERATOR
|--SCAN new_stats
`--SEARCH strds_metadata USING COVERING INDEX sqlite_autoindex_strds_metadata_1 (id=?)
```
And this is the QUERY plan with a ` JOIN`:
```
QUERY PLAN
|--MATERIALIZE new_stats
|  |--USE TEMP B-TREE FOR count(DISTINCT)
|  |--SCAN raster_map_register_0451659289d64936b320841a3c074dda
|  `--SEARCH raster_metadata USING INDEX sqlite_autoindex_raster_metadata_1 (id=?)
|--SCAN new_stats
`--SEARCH strds_metadata USING COVERING INDEX sqlite_autoindex_strds_metadata_1 (id=?)
```

So I did a test and ran the query alternatives for strds metadata 50 times on a 5000 maps STRDS. There I could not notice any performace difference (0.011.- 0.020 s) After registering another 5000 maps, these was still no clear winner, but the JOIN had lower maximum run times. So I think I will stick with the JOIN approach...

Reply:
Do you know how I could get a sqlite DB that I could test some queries on (ideally independent of grass)? Is it really big of confidential that it can't be attached to this issue? Both of these query plans don't seem ideal (the use of SCAN especially, but also autoindex), but it really kind of depends on the normal size of these. If you end up being able to refer one to me, what are the values of the placeholders that get replaced?